test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1386 @@
+                "iprange",
+                "--dst-range",
+                "10.0.0.50-10.0.0.100",
+            ],
+        ),
+    ],
+)
+def test_iprange(mocker, mock_basic_commands, test_input, expected):
+    """Test iprange module with its flags src_range and dst_range."""
+    set_module_args(test_input)
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == expected
+
+
+def test_insert_rule_with_wait(mocker, mock_basic_commands):
+    """Test flush without parameters."""
+    set_module_args(
+        {
+            "chain": "OUTPUT",
+            "source": "1.2.3.4/32",
+            "destination": "7.8.9.10/42",
+            "jump": "ACCEPT",
+            "action": "insert",
+            "wait": "10",
+        }
+    )
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "OUTPUT",
+        "-w",
+        "10",
+        "-s",
+        "1.2.3.4/32",
+        "-d",
+        "7.8.9.10/42",
+        "-j",
+        "ACCEPT",
+    ]
+
+
+def test_comment_position_at_end(mocker, mock_basic_commands):
+    """Test comment position to make sure it is at the end of command."""
+    set_module_args(
+        {
+            "chain": "INPUT",
+            "jump": "ACCEPT",
+            "action": "insert",
+            "ctstate": ["NEW"],
+            "comment": "this is a comment",
+            "_ansible_check_mode": True,
+        }
+    )
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "INPUT",
+        "-j",
+        "ACCEPT",
+        "-m",
+        "conntrack",
+        "--ctstate",
+        "NEW",
+        "-m",
+        "comment",
+        "--comment",
+        "this is a comment",
+    ]
+    assert run_command.call_args[0][0][14] == "this is a comment"
+
+
+def test_destination_ports(mocker, mock_basic_commands):
```

Comment:
It seems to me that more test functions can be collapsed into the parametrized test.

Reply:
Was this case non-parametrizable somehow?

Reply:
I would keep them as is they might be specific to the use case.